x-pack/plugins/apm/common/anomaly_detection/apm_ml_detectors.ts

```diff
@@ -5,8 +5,28 @@
  * 2.0.
  */
 
-export const enum ApmMlDetectorIndex {
-  txLatency = 0,
-  txThroughput = 1,
-  txFailureRate = 2,
+export const enum ApmMlDetectorType {
+  txLatency = 'txLatency',
+  txThroughput = 'txThroughput',
+  txFailureRate = 'txFailureRate',
+}
+
+const detectorIndices = {
+  [ApmMlDetectorType.txLatency]: 0,
+  [ApmMlDetectorType.txThroughput]: 1,
+  [ApmMlDetectorType.txFailureRate]: 2,
+};
+
+export function getApmMlDetectorIndex(type: ApmMlDetectorType) {
+  return detectorIndices[type];
+}
+
+export function getApmMlDetectorType(detectorIndex: number) {
+  let type: ApmMlDetectorType;
+  for (type in detectorIndices) {
+    if (detectorIndices[type] === detectorIndex) {
+      return type;
+    }
+  }
+  throw new Error('Could not map detector index to type');
```

Comment:
Can this be simplified?

```suggestion
  return Object.keys(detectorIndices)[detectorIndex];
```

Reply:
it'd depend on whatever order keys is returned in. I know that should be OK in most browsers, but I'd prefer the explicitness of a for loop.

Reply:
Ah, true. The old unstable ordering of objects. Pretty sure objects are stable nowadays but I agree, better safe than sorry.

Still, WDYT of this?

```suggestion
  return Object.entries(detectorIndices).find(([name, index]) => index === detectorIndex)
```